lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -0,0 +1,91 @@
+"""
+Serializers for LTI-related endpoints
+"""
+import six
+
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import UsageKey
+
+from lti_consumer.models import LtiAgsLineItem
+
+
+class UsageKeyField(serializers.Field):
+    """
+    Serializer field for a model UsageKey field.
+
+    Recreated here since we cannot import directly from
+    from the platform like so:
+    `from openedx.core.lib.api.serializers import UsageKeyField`
+    """
+    # pylint: disable=arguments-differ
+    def to_representation(self, data):
+        """Convert a usage key to unicode. """
+        return six.text_type(data)
+
+    def to_internal_value(self, data):
+        """Convert unicode to a usage key. """
+        try:
+            return UsageKey.from_string(data)
+        except InvalidKeyError:
+            raise serializers.ValidationError(u"Invalid usage key: {}".format(data))
+
+
+class LtiAgsLineItemSerializer(serializers.ModelSerializer):
+    """
+    LTI AGS LineItem Serializer.
+
+    This maps out the internally stored LineItemParameters to
+    the LTI-AGS API Specification, as shown in the example
+    response below:
+
+    {
+        "id" : "https://lms.example.com/context/2923/lineitems/1",
+        "scoreMaximum" : 60,
+        "label" : "Chapter 5 Test",
+        "resourceId" : "a-9334df-33",
+        "tag" : "grade",
+        "resourceLinkId" : "1g3k4dlk49fk"
+        "startDateTime": "2018-03-06T20:05:02Z"
+        "endDateTime": "2018-04-06T22:05:03Z"
+    }
+
+    Note: The platform MUST NOT modify the 'resourceId', 'resourceLinkId' and 'tag' values.
```

Comment:
@nedbat That was a straight copy from the IMS LTI-AGS spec ([here](https://www.imsglobal.org/spec/lti-ags/v2p0#creating-a-new-line-item)). I've moved this to the model since it makes more sense to have it there.